### PR TITLE
make `bytecheck::Error` Send & Sync

### LINKED ISSUE #25: `CheckBytes::Error` should be `Send + Sync`
Migrated from https://github.com/rkyv/rkyv/issues/320. Bounding the error type in this way is a breaking change but will greatly improve compatibility with other error handling frameworks like anyhow and thiserror.

### PR DESCRIPTION
**Pull request by mkatychev**
Resolves: https://github.com/rkyv/bytecheck/issues/25

### CODE REVIEW

**Comment by mkatychev** [comment]
Not sure if this should be `Send + Sync` as well:
https://github.com/rkyv/bytecheck/blob/2f241f648229ddd3a266723588fde96d4f0678dd/bytecheck/src/lib.rs#L183-L187

**Comment by djkoloski** [comment]
Yeah, I think we should add `Send + Sync` to that as well. That will make bytecheck's error type compatible with anyhow.

File: bytecheck/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ readme = "crates-io.md"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-bytecheck_derive = { version = "=0.6.9", path = "../bytecheck_derive", default-features = false }
+bytecheck_derive = { version = "=0.7.0", path = "../bytecheck_derive", default-features = false }
```
Comment (mkatychev) [inline bytecheck/Cargo.toml:17]: should this be workspace versioned for the workspace crates in this repo? 

```toml
[package]
name = "bytecheck"
version.workspace = true
edition.workspace = true
```

Comment (djkoloski) [reply bytecheck/Cargo.toml:17]: Probably yeah. If you want, a follow-up PR would be ideal.

**Pull request merged**